speechbrain/lobes/models/transformer/Conformer.py

```diff
@@ -62,6 +62,10 @@ class ConformerEncoderLayerStreamingContext:
 class ConformerEncoderStreamingContext:
```

Comment:
Will we need a definition of this context for every single class we want to use? Can't we just have a generic one? Overall, having model-dependent code in a toolkit that may have hundreds of models is not a good practice imho.

Reply:
I don't think it can be made generic unless you can think of a cleaner way. I defined the streaming context objects this way largely because the data that needs to be memorized is highly module-specific (it might not even be tensors), and because it needed to be recursive in some way.

Anything that requires to be aware of streaming needs to define what data they need to hold *at some point*, so I don't think it can theoretically be improved much.

If a module knows it never actually needs to save anything for streaming, then it doesn't need to actually define a context class. (And if a module that contains it calls into `.make_streaming_context` anyway, it can just return `None`.)

I think the ugliest part is carrying along the training configuration down the hierarchy, but that's also kind of the only thing that needs to be this way.

Reply:
I changed my mind on this, it's ok.